#! /usr/bin/env python

PACKAGE='draco_point_cloud_transport'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# TODO: Draco config
gen.add("encode_speed", int_t,   0, "0 = slowest speed, but the best compression 10 = fastest, but the worst compression.",  7, 0, 10)
gen.add("decode_speed", int_t, 0, "0 = slowest speed, but the best compression 10 = fastest, but the worst compression.",  7, 0, 10)
gen.add("quantization_POSITION", int_t, 0, "Number of bits for quantization of POSITION type attributes.",  14, 8, 31)
gen.add("quantization_NORMAL",  int_t, 0, "Number of bits for quantization of NORMAL type attributes.",  14, 8, 31)
gen.add("quantization_COLOR",  int_t, 0, "Number of bits for quantization of COLOR type attributes.",  14, 8, 31)
gen.add("quantization_TEX_COORD",  int_t, 0, "Number of bits for quantization of TEX_COORD type attributes.",  14, 8, 31)
gen.add("quantization_GENERIC",  int_t, 0, "Number of bits for quantization of GENERIC type attributes.",  14, 8, 31)
gen.add("method", int_t, 0, "Encoding process method 0 = KD-tree 1 = sequential", 0, 0, 1)


exit(gen.generate(PACKAGE, "DracoPublisher", "DracoPublisher"))
